burn-common/src/id.rs

```diff
@@ -0,0 +1,78 @@
+use alloc::string::{String, ToString};
+
+use crate::rand::{get_seeded_rng, Rng, SEED};
+
+use uuid::{Builder, Bytes};
+
+pub struct IdGenerator {}
+
+impl IdGenerator {
+    pub fn generate() -> String {
+        let mut seed = SEED.lock().unwrap();
+        let mut rng = if let Some(rng_seeded) = seed.as_ref() {
+            rng_seeded.clone()
+        } else {
+            get_seeded_rng()
+        };
+
+        let random_bytes: Bytes = rng.gen();
+        *seed = Some(rng);
+
+        let uuid = Builder::from_random_bytes(random_bytes).into_uuid();
+
+        uuid.as_hyphenated().to_string()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use alloc::{collections::BTreeSet, string::String};
+
+    #[cfg(feature = "std")]
+    use dashmap::DashSet; //Concurrent HashMap
+    #[cfg(feature = "std")]
+    use std::{sync::Arc, thread};
+
+    #[test]
+    fn not_empty_test() {
+        assert_eq!(IdGenerator::generate().is_empty(), false);
+    }
+
+    #[test]
+    fn uniqueness_test() {
+        let mut set: BTreeSet<String> = BTreeSet::new();
+
+        const IDS_CNT: usize = 10_000;
+        for _i in 0..IDS_CNT {
+            assert_eq!(set.insert(IdGenerator::generate()), true);
+        }
+
+        assert_eq!(set.len(), IDS_CNT);
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn thread_safety_test() {
```

Comment:
I would extract `10` and `1000` and `10_000` in variables, like `num_threads` , `num_repeat` and `num_added`.

Reply:
OK